### ISSUE TITLE
Autocorrect produces invalid syntax for Style/TrailingCommaIn(Array|Hash)Literal and Layout/Multiline(Array|Hash)BraceLayout

### ISSUE THREAD

**Issue by labocho** [rubocop-hq/rubocop#7271]
In some case below, auto-correction may produce invalid syntax.

- `Style/TrailingCommaInArrayLiteral` or  `Style/TrailingCommaInHashLiteral` is configured with `EnforcedStyleForMultiline: consistent_comma`.
- Enable `Style/TrailingCommaIn(Array|Hash)Literal` and `Layout/Multiline(Array|Hash)BraceLayout` at same time.

--------

## Expected behavior

Do not change semantics on auto-correct.

## Actual behavior

Change semantics on auto-correct.

## Steps to reproduce the problem

```yaml
# .rubocop.yml
Style/TrailingCommaInArrayLiteral:
  EnforcedStyleForMultiline: consistent_comma

Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: consistent_comma
```

### Array literal

```ruby
# multi_line_array.rb
[ 1,
  2
].to_s
```

`rubocop --only=Style/TrailingCommaInArrayLiteral,Layout/MultilineArrayBraceLayout -a multi_line_array.rb` produces invalid source as below.

```ruby
[ 1,
  2],.to_s
```

### Hash literal

```ruby
# multi_line_hash.rb
{ foo: 1,
  bar: 2
}.to_s
```
`rubocop --only=Style/TrailingCommaInHashLiteral,Layout/MultilineHashBraceLayout -a multi_line_hash.rb` produces invalid source as below.

```ruby
{ foo: 1,
  bar: 2},.to_s
```

## RuboCop version

```
$ rubocop -V
0.74.0 (using Parser 2.6.3.0, running on ruby 2.6.2 x86_64-darwin18)
```

**Comment by koic** [comment]
Dup #6498.

**Comment by buehmann** [comment]
I understand what is happening and have a fix for it. @koic How do I best add a spec that verifies that these pairs of cops work correctly together? Is there a best practice for this?

**Comment by koic** [comment]
@buehmann Tests for auto-correction with multiple cops can be written to spec/rubocop/cop/layout/indentation_width_spec.rb. The following address may be helpful.
https://github.com/rubocop-hq/rubocop/pull/6866/files#diff-0e7dbdd6e99e0defc2bab2cb544b42da

**Comment by labocho** [comment]
Thanks!
